tck/mp-jwt-embedded/src/main/java/org/apache/tomee/microprofile/jwt/cdi/ClaimBean.java

```diff
@@ -0,0 +1,361 @@
+import javax.enterprise.inject.spi.Annotated;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.inject.spi.PassivationCapable;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import javax.json.bind.Jsonb;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+
+@Vetoed
+public class ClaimBean<T> implements Bean<T>, PassivationCapable {
+
+    private static Logger logger = Logger.getLogger(MPJWTCDIExtension.class.getName());
+
+    private final static Set<Annotation> QUALIFIERS = new HashSet<>();
+
+    static {
+        QUALIFIERS.add(new ClaimLiteral());
+    }
+
+    @Inject
+    private Jsonb jsonb;
+
+    @Inject
+    private JsonWebToken jsonWebToken;
+
+    private final BeanManager bm;
+    private final Class rawType;
+    private final Set<Type> types;
+    private final String id;
+    private final Class<? extends Annotation> scope;
+
+    public ClaimBean(final BeanManager bm, final Type type) {
+        this.bm = bm;
+        types = new HashSet<>();
+        types.add(type);
+        rawType = getRawType(type);
+        this.id = "ClaimBean_" + types;
+        scope = Dependent.class;
+    }
+
+    private Class getRawType(final Type type) {
+        if (type instanceof Class) {
+            return (Class) type;
+
+        } else if (type instanceof ParameterizedType) {
+            final ParameterizedType paramType = (ParameterizedType) type;
+            return (Class) paramType.getRawType();
+        }
+
+        throw new UnsupportedOperationException("Unsupported type " + type);
+    }
+
+
+    @Override
+    public Set<InjectionPoint> getInjectionPoints() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Class<?> getBeanClass() {
+        return rawType;
+    }
+
+    @Override
+    public boolean isNullable() {
+        return false;
+    }
+
+    @Override
+    public void destroy(T instance, CreationalContext<T> context) {
+
+    }
+
+    @Override
+    public Set<Type> getTypes() {
+        return types;
+    }
+
+    @Override
+    public Set<Annotation> getQualifiers() {
+        return QUALIFIERS;
+    }
+
+    @Override
+    public Class<? extends Annotation> getScope() {
+        return scope;
+    }
+
+    @Override
+    public String getName() {
+        return null;
+    }
+
+    @Override
+    public Set<Class<? extends Annotation>> getStereotypes() {
+        return Collections.emptySet();
```

Comment:
Is this definitely expected to be immutable?

Reply:
Yes